src/firmware/app/control/trajectory_planner_test.cpp

```diff
@@ -0,0 +1,529 @@
+    path_parameters.path                       = path;
+    path_parameters.t_start                    = 0;
+    path_parameters.t_end                      = 1;
+    path_parameters.num_segments               = 5999;
+    path_parameters.max_allowable_acceleration = 3;
+    path_parameters.max_allowable_speed        = 3;
+    path_parameters.initial_speed              = 0;
+    path_parameters.final_speed                = 0;
+
+    TrajectoryElement_t const_arc_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    Trajectory_t trajectory = {.trajectory_elements = const_arc_elements};
+
+
+    app_trajectory_planner_generateConstantArcLengthSegmentation(path_parameters,
+                                                                 &trajectory);
+
+    // Create the parameterization to contain the desired number of segments
+    CREATE_STATIC_ARC_LENGTH_PARAMETRIZATION(arc_length_param,
+                                             TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS);
+
+    // Get all of the points for the arc length parameterization (Not constant arc length
+    // segments)
+    shared_polynomial_getArcLengthParametrizationOrder3(
+        path, path_parameters.t_start, path_parameters.t_end, arc_length_param);
+
+    Trajectory_t const_interp_trajectory;
+    TrajectoryElement_t const_interp_elements[TRAJECTORY_PLANNER_MAX_NUM_ELEMENTS];
+    const_interp_trajectory.trajectory_elements = const_interp_elements;
+
+    // Calculate the constant-interpolation period equivalent of the trajectory
+    app_trajectory_planner_interpolateConstantTimeTrajectorySegmentation(
+        &const_interp_trajectory, &trajectory, 0.001);
+
+    EXPECT_NEAR(trajectory.trajectory_elements[trajectory.num_elements - 1].position.x,
+                const_interp_trajectory
+                    .trajectory_elements[const_interp_trajectory.num_elements - 1]
+                    .position.x,
+                0.001);
+    EXPECT_NEAR(trajectory.trajectory_elements[trajectory.num_elements - 1].position.y,
+                const_interp_trajectory
+                    .trajectory_elements[const_interp_trajectory.num_elements - 1]
+                    .position.y,
+                0.001);
+    EXPECT_NEAR(trajectory.trajectory_elements[trajectory.num_elements - 1].time,
+                const_interp_trajectory
+                    .trajectory_elements[const_interp_trajectory.num_elements - 1]
+                    .time,
+                0.001);
+
+    EXPECT_NEAR(trajectory.trajectory_elements[0].position.x,
+                const_interp_trajectory.trajectory_elements[0].position.x, 0.001);
+    EXPECT_NEAR(trajectory.trajectory_elements[0].position.y,
+                const_interp_trajectory.trajectory_elements[0].position.x, 0.001);
+    EXPECT_NEAR(trajectory.trajectory_elements[0].time,
+                const_interp_trajectory.trajectory_elements[0].time, 0.001);
+}
+
+TEST_F(TrajectoryPlannerTest, test_assert_cannot_reach_final_velocity)
```

Comment:
Can we add a test that checks we get `TrajectoryPlannerGenerationStatus::OK` if generation succeeds?

@garethellis0 preference on if this is its own test or added to existing ones with "valid dynamics"? Because this code is more C-style not as confident as to the best convention.

Reply:
Added the check for OK to all trajectories that are supposed to be valid